vulkan_generator/generator.py

```diff
@@ -57,8 +61,32 @@ def print_vulkan_metaadata(vulkan_metadata: types.AllVulkanTypes) -> None:
     pretty_printer.pprint(vulkan_metadata.funcpointers)
 
 
-def generate(vulkan_xml_path: Path) -> bool:
+def basic_generate(target: str,
+                   output_dir: Path,
+                   all_vulkan_types: types.AllVulkanTypes,
+                   generate_header,
+                   generate_cpp,
+                   generate_test):
+
+    generate_header(os.path.join(output_dir, target + ".h"), all_vulkan_types)
+    generate_cpp(os.path.join(output_dir, target + ".cpp"), all_vulkan_types)
+    generate_test(os.path.join(output_dir, target + "_tests.cpp"), all_vulkan_types)
+
+def generate(target: str, output_dir: Path, vulkan_xml_path: Path) -> bool:
+
     """ Generator function """
     all_vulkan_types = vulkan_parser.parse(vulkan_xml_path)
     print_vulkan_metaadata(all_vulkan_types)
+
+    os.makedirs(output_dir, exist_ok=True)
+
+    """ Switch table for generate target. Add new targets here and throw exception for unknown targets """
+    if target == "HandleRemapper":
```

Comment:
I think it would be better if each generator is its own binary instead of having a master binary with a "generate <kind> code" flag. When changing something in the code generator, if there are separate generators, only the one that actually changed gets executed, otherwise all code would have to be regenerated since the "master generator" has changed from the POV of bazel.

Reply:
I will leave Melih's vulkan generator for now as if I remove this generation from his generator, it doesn't generate anything at all and has no reason to exist beyond testing. Lets make this follow up work. Created b/236110214 to track this and assigned the work to myself.